seagoat/repository.py

```diff
@@ -92,6 +92,9 @@ def analyze_files(self):
             "--no-merges",
         ]
 
+        if (max_commits := self.config["server"]["readMaxCommits"]) is not None:
```

Comment:
looks great, but I would add a test :pray: 

I think in this case it's best not to think hard about how to do an integration test, it's  enough to make a simple unit test that makes sure that the correct value is appended to the command.

you can also refactor the code a bit if you want to avoid mocking or simplify the unit test (such as moving the logic that creates the command into a utility function or a private method)